src/test/java/de/retest/recheck/ignore/SearchFilterFilesTest.java

```diff
@@ -135,6 +136,25 @@ void getFilterByName_should_return_user_filter_with_given_name_over_default_filt
 		assertThat( result.toString() ).isEqualTo( "CompoundFilter(filters=[])" );
 	}
 
+	@Test
+	@ClearSystemProperty( key = RETEST_PROJECT_ROOT )
+	void getFilterByName_should_only_load_searched_filter() throws IOException {
+		System.setProperty( RETEST_PROJECT_ROOT, filterFolder.toString() );
+
+		// Create infinite filter loop which will cause infinite loop if loaded
+		final Path resolve = filterFolder.resolve( "foo.filter" );
```

Comment:
Since you named below `bar`:

```suggestion
		final Path foo = filterFolder.resolve( "foo.filter" );
```

Although `bar` appears to be never used?

Reply:
That was copy-paste error. I fixed that.